backend/src/views.py

```diff
@@ -95,38 +97,56 @@ def delete_rule(request, rule_id, userid):
     rule.delete()
     return Response(None, status=status.HTTP_204_NO_CONTENT)
 
+def get_transaction_formatted_rule_list(userid):
+    rules = Rule.objects.filter(userid=userid)
+    rrules = RuleSerializer(rules, many=True).data
+    jsonBodyForTransactionQuery = {}
+
+    for rrule in rrules:
+        #TODO Check for duplicate keys and reconcile.
+        jsonBodyForTransactionQuery[rrule['name']] = {
```

Comment:
Put a TODO to switch to 'id' instead of 'name' when the UI is ready for it.

Reply:
Done